### ISSUE TITLE
Elementor Pro - Beta3 - Archive Issue

### ISSUE THREAD

**Issue by stickypages** [pojome/elementor#3996]
I'm going to do my best to explain this issue. But it is rather confusing, sorry. 

Short story: The Archive templates are not using the content structure supplied. 
I have a theme, that have a wrapper, something like this.

```
<div class="Left-Col">
   <!-- NAV -->
</div>
<div class="MainContent-Col">
  <!-- Main content  the_content(); --> 
  <?php 
               if ( is_archive() || is_home() || is_tag() ||  is_search() ) {
			if ( ! $has_elementor || ! elementor_theme_do_location( 'archive' ) ) {
				do_action('theme_archive');
			}
		}
   ?>
</div>
```

So this works for all the other template options I have setup. Except for the Archive. And the terminology "Override" is somewhat confusing here. What it is outputting now is removing the MainContent-Col.

```
<div class="Left-Col">
   <!-- NAV -->
</div>
<div class="Elementor-Element">
  <!-- Archive --->
</div>

```

Because of it changing the main structure, it breaks the site. I have tried setting it up with it's own Archive.PHP file, but it doesn't seem to want to keep my structure I set.... and the term "Override" I am assuming all this. 

What's the best setup for Archive Templates then? Thanks for your time.

I have a few screenshots to help support. 
![2018-04-03 21_24_54-deck - rykon construction](https://user-images.githubusercontent.com/658488/38288756-75c0169c-3787-11e8-8fea-32a8895c16ce.png)
![2018-04-03 21_23_12-my templates rykon construction wordpress](https://user-images.githubusercontent.com/658488/38288757-75d97a74-3787-11e8-89d3-1b1617a785e2.png)
![2018-04-03 21_41_49-deck - rykon construction](https://user-images.githubusercontent.com/658488/38288866-083831bc-3788-11e8-9a43-5d1bcac89bd0.png)
![2018-04-03 21_41_37-deck - rykon construction](https://user-images.githubusercontent.com/658488/38288867-084f9a32-3788-11e8-9159-e297dcf31678.png)

**Comment by stickypages** [comment]
I noticed an additional issue here. I was reading deeper into some other issues from other people. When I enable all theme locations using:
```
add_action( 'elementor/theme/register_locations', function($elementor_theme_manager) {
	$elementor_theme_manager->register_all_core_location();
});
```

It didn't enable my 'archive' setup. So, when I add the following, it seems to remove the 'override' text....
```
add_action( 'elementor/theme/register_locations', function($elementor_theme_manager) {
	$elementor_theme_manager->register_all_core_location();
	$elementor_theme_manager->register_location( 'archive' );
});
```

Seems to be all working now as far as I can tell....

**Comment by ramiy** [comment]
**Locations Registration:**

you can either register all the core locations at once:

```
function theme_prefix_register_elementor_locations( $elementor_theme_manager ) {

	$elementor_theme_manager->register_all_core_location();

}
add_action( 'elementor/theme/register_locations', 'theme_prefix_register_elementor_locations' );
```

or register the core locations one-by-one:

```
function theme_prefix_register_elementor_locations( $elementor_theme_manager ) {

	$elementor_theme_manager->register_location( 'header' );
	$elementor_theme_manager->register_location( 'footer' );
	$elementor_theme_manager->register_location( 'single' );
	$elementor_theme_manager->register_location( 'archive' );

}
add_action( 'elementor/theme/register_locations', 'theme_prefix_register_elementor_locations' );
```

Try the second way if the first way doesn't work.

**Comment by stickypages** [comment]
@ramiy 
Yes, I did try the second one like I mentioned and it did seem to work better.
The _missing var_, was just a different representation and is just a var that holds the 'function exists'. 

On the implementation, I switched from 'actions' to full Function() and will see how this affects things.
